build.rs

```diff
@@ -269,51 +346,50 @@ fn main() {
     // Avoid assuming the working directory is the same is the $CARGO_MANIFEST_DIR so that toolchains
     // which may assume other working directories can still build this code.
     let c_root_dir = PathBuf::from(
-        env::var_os("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should always be set"),
+        env::var_os(&env::CARGO_MANIFEST_DIR).expect("CARGO_MANIFEST_DIR should always be set"),
     );
 
     // Keep in sync with `core_name_and_version!` in prefixed.rs.
     let core_name_and_version = [
-        &env::var("CARGO_PKG_NAME").unwrap(),
+        &env::var(&env::CARGO_PKG_NAME).unwrap(),
         "core",
-        &env::var("CARGO_PKG_VERSION_MAJOR").unwrap(),
-        &env::var("CARGO_PKG_VERSION_MINOR").unwrap(),
-        &env::var("CARGO_PKG_VERSION_PATCH").unwrap(),
-        &env::var("CARGO_PKG_VERSION_PRE").unwrap(), // Often empty
+        &env::var(&env::CARGO_PKG_VERSION_MAJOR).unwrap(),
+        &env::var(&env::CARGO_PKG_VERSION_MINOR).unwrap(),
+        &env::var(&env::CARGO_PKG_VERSION_PATCH).unwrap(),
+        &env::var(&env::CARGO_PKG_VERSION_PRE).unwrap(), // Often empty
     ]
     .join("_");
     // Ensure `links` in Cargo.toml is consistent with the version.
     assert_eq!(
-        &env::var("CARGO_MANIFEST_LINKS").unwrap(),
+        &env::var(&env::CARGO_MANIFEST_LINKS).unwrap(),
         &core_name_and_version
     );
 
-    const RING_PREGENERATE_ASM: &str = "RING_PREGENERATE_ASM";
-    match env::var_os(RING_PREGENERATE_ASM).as_deref() {
+    match env::var_os(&env::RING_PREGENERATE_ASM).as_deref() {
         Some(s) if s == "1" => {
             pregenerate_asm_main(&c_root_dir, &core_name_and_version);
         }
         None => ring_build_rs_main(&c_root_dir, &core_name_and_version),
         _ => {
-            panic!("${} has an invalid value", RING_PREGENERATE_ASM);
+            panic!("RING_PREGENERATE_ASM has an invalid value",);
```

Comment:
NIT: I don't think it makes sense to keep the trailing comma.

But, I think you can instead use `RING_PREGENERATE_ASM.name` as the parameter instead, to DRY this.

Reply:
Fixed.